code/mathematical_algorithms/src/coprime_numbers/coprime_numbers.java

```diff
@@ -0,0 +1,27 @@
+import java.io.InputStreamReader;
+import java.util.Scanner;
+
+class CoprimeNumbers {
+	public static int gcd(int num1 , int num2) {
+		int tmp = 1;
+		while(tmp != 0) {
```

Comment:
`while true` is fine. Otherwise, this loop may return nothing.

Reply:
I tried many test cases it works. That scenario of no return will not be an issue.